gitpod-io/gitpod
# Add insights page

## Description

Adds a `/insights` page, offering access to the listing from `WorkspaceService.ListWorkspaceSessions` both as a table and as a downloadable CSV, similar to the one under `/usage`.

<img width="1473" alt="image" src="https://github.com/user-attachments/assets/b9e13fc5-850f-4177-a3aa-c5e646c84b7e" />

## Related Issue(s)

Fixes CLC-1017

## How to test

1. [Join my org](https://ft-add-insights-page.preview.gitpod-dev.com/orgs/join?inviteId=65761649-2cff-4163-a378-3e93b9733a13) (to get some initial data)
2. (optional) start a workspace
3. Check out `/insights`, along with the downloadable CSV

## Documentation

Possibly, yeah. It's not as crucial now, though, since it's not publicly accessible (without knowing about the endpoint) and also pretty straight forward as far as interaction goes.
